src/Injector.php

```diff
@@ -145,11 +138,10 @@ public function create(string $name, array $parameters = [])
      *
      * @param string $name The type name to instantiate.
      * @param array  $params Constructor arguments, keyed by the parameter name.
-     * @return object
      * @throws InvalidCallbackException
      * @throws ClassNotFoundException
      */
-    protected function createInstance(string $name, array $params)
+    protected function createInstance(string $name, array $params): object
```

Comment:
this is also BC break when extended class override it, ref https://3v4l.org/lldZ7

Reply:
I see now, thanks again for the insight. 
Sorry for my lack of knowledge and wasting your and @Ocramius time but I think this pull request can be closed/ignored then.